model/system_metrics.go

```diff
@@ -32,6 +32,27 @@ var (
 	systemMetricsArtifactKey    = bsonutil.MustHaveTag(SystemMetrics{}, "Artifact")
 )
 
+// CreateSystemMetrics is the entry point for creating the metadata for
+// system metric time series data for a task execution.
+func CreateSystemMetrics(info SystemMetricsInfo, artifact SystemMetricsArtifactInfo) *SystemMetrics {
```

Comment:
Don't pass in the artifact here, since we are creating it there is nothing to pass in from the user.